cgsmiles/sample.py

```diff
@@ -10,17 +11,134 @@
+    >>> polymer_reactivites = {">A":0.5, "<A":0, ">B": 0.5, "<B": 0.0}
+    >>> fragment_reactivies = {'$A': {'$A': 0.,   '$C': 0.,   '$B': 0.7, '$D': 0.3},
+                               '$B': {'$A': 0.7, '$C': 0.3, '$B': 0.0, '$D': 0.0 },
+                               '$C': {'$A': 0.,   '$C': 0.,   '$B': 0.3, '$D': 0.7},
+                               '$D': {'$A': 0.3, '$C': 0.7, '$B': 0.0, '$D': 0.0 }},
+    >>> sampler = MoleculeSampler.from_string(cgsmiles_str,
+                                              polymer_reactivies=polymer_reactivies,
+                                              fragment_reactivies=fragment_reactivies)
+    >>> polymer_graph = sampler.sample(target_weight=100000)
+
+    Advanced API Examples
+    ---------------------
+    Of course the sampler is not limited to linear co-polymers. Also randomly
+    branched molecules such as bottle-brush copolymers can be generated. To
+    steer the branching ratio and branch extension as well as terminal end
+    groups the `branch_term_probs`, `terminal_fragments`, and `terminal_reactivities`
+    can be provided.
+
+    For example, To generate a bottle brush polymer that has PMA in the backbone
+    and PEG as side-chain terminated with an OH group the following CGSmiles string
+    in combination with the above mentioned probabilities can be provided.
+
+    >>> cgsmiles_str="{#PMA=[>]CC[<]C(=O)OC[>A],#PEG=[<A]COC[>A],#OH=[<A]O}"
+    >>> sampler = MoleculeSampler.from_fragment_string(cgsmiles_str,
+                                                       branch_term_probs={"PEG": 0.3, "PMA":0.0},
+                                                       terminal_fragments=['OH'],
+                                                       bonding_probabilities={'<': 0.01, '>': 0.01, '>A': 0.99, '<A':0.99},
+                                                       bond_term_probs={'<A': 1., '>A':1., '>': 0, '<': 0},
+                                                       all_atom=True)
+    >>> molecule = sampler.sample(target_weight=5008)
+
+    This combination results into a dense brush where very PMA molecule has a PEG
+    branch whose length is determined by a 1 in 3 probability to terminate.
+
+    In contrast one can also generate spares branched molecules. For instance,
+    Dextran at low molecular weights is mainly a linear alpha 1-6 polyglucose
+    polymer. Occasionally, a 1-4 branch extends from it which has lengths of 2-4.
+    At the Martini coarse-grained level Dextran is described by three particles
+    A, B, and C. The alpha 1-6 bonds are between A and C whereas the 1-4 bonds
+    are between A and B.
+
+    >>> cgsmiles_str="{#GLC=[$A][#A]1[#B][$B][#C]1[$C]}"
+    >>> sampler = MoleculeSampler.from_fragment_string(cgsmiles_str,
+                                                       fragment_masses={'GLC': 165},
+                                                       bonding_probabilities={'$A': 0.8, '$C': 0.2, '$B': 0.2},
```

Comment:
This does not sum to 1?

Reply:
does not have to but I guess might be better for the example purpose.